### ISSUE TITLE
Sockets not closed when body is not consumed

### ISSUE THREAD

**Issue by tripodsan** [adobe/helix-fetch#32]
**Description**
it looks like the sockets are not closed when the body is not consumed; despite a call to disconnect all.

**To Reproduce**

```js
const fetchAPI = require('@adobe/helix-fetch');

async function request(context, error) {
  const resp = await context.fetch('https://adobeioruntime.net/api/v1/web/helix/helix-services/word2md@v1?path=%2Fdefault.md&shareLink=https%3A%2F%2Fadobe.sharepoint.com%2Fsites%2FTheBlog%2FShared%2520Documents%2Ftheblog&rid=VqNCOOblZXzBlnpLTvgG39uWoAIrGDWF&src=adobe%2Ftheblog%2Fdd25127aa92f65fda6a0927ed3fb00bf5dcea069', {
    headers: {
      accept: 'application/json',
    }
  });
  if (error) {
    throw error;
  }
  const text = await resp.text();
  console.log(resp.status, text);
}

async function run() {
  const context = fetchAPI.context({});
  try {
    await request(context);
    await request(context, new Error('user error'));
  } finally {
    await context.disconnectAll();
  }
}
run().catch(console.error);
```

```console
$ node src/simple.js
404 Error while converting document
Error: user error
    at run (src/simple.js:36:28)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
...(process hangs)...
```

**Note**

adding `keepAlive: false` seems to solve the problem:

```js
  const context = fetchAPI.context({
    http1: {
      keepAlive: false,
    },
  });
```

**Comment by stefan-guggisberg** [comment]
Strange, I've reported this issue a while ago: https://github.com/grantila/fetch-h2/issues/77

Supposedly it had been fix. Apparently that's not the case.

**Comment by stefan-guggisberg** [comment]
I have verified that the problem described in grantila/fetch-h2#77 does indeed still exist. I have created a new issue: https://github.com/grantila/fetch-h2/issues/101

It is specific to HTTP1(.1) connections, i.e. with HTTP2 the problem doesn't occur. It is also independent of the response status code. 

The problem is somehow hidden/mitigated by `helix-fetch` due to caching. The body of cached responses are always consumed.  Non-cachable responses however exhibit this problem.

The following code reproduces the issue with `helix-fetch`:

```js
const { fetch, disconnectAll } = require('@adobe/helix-fetch');

(async function () {
  const resp = await fetch('http://httpbin.org/status/200', { cache: 'no-store' });
  await disconnectAll();
}());
```

It leaves an open socket despite calling `disconnectAll()` which prevents the process from exiting.
